qiskit_superstaq/custom_gates.py

```diff
@@ -13,77 +13,79 @@ class AceCR(qiskit.circuit.Gate):
+        self,
+        rads: Union[str, float] = np.pi / 2,
+        sandwich_rx_rads: float = 0,
+        label: Optional[str] = None,
     ) -> None:
-        if polarity not in ("+-", "-+"):
+        if rads == "+-":
+            rads = np.pi / 2
+        elif rads == "-+":
+            rads = -np.pi / 2
+        elif isinstance(rads, str):
             raise ValueError("Polarity must be either '+-' or '-+'")
 
-        name = "acecr_" + polarity.replace("+", "p").replace("-", "m")
-
+        name = "acecr"
+        params = [rads]
         if sandwich_rx_rads:
-            super().__init__(name + "_rx", 2, [sandwich_rx_rads], label=label)
-        else:
-            super().__init__(name, 2, [], label=label)
-
-        self.polarity = polarity
+            name += "_rx"
+            params.append(sandwich_rx_rads)
+        super().__init__(name, 2, params, label=label)
+
+        rads_str = f"{rads}"
+        if np.isclose(round(rads / np.pi, 5) * np.pi, rads):
+            rads_str = f"{round(rads / np.pi, 5)}π"
+        self.rads = rads
+        self._rads_str = rads_str
         self.sandwich_rx_rads = sandwich_rx_rads
 
     def inverse(self) -> "AceCR":
-        return AceCR(self.polarity, sandwich_rx_rads=-self.sandwich_rx_rads, label=self.label)
+        return AceCR(self.rads, -self.sandwich_rx_rads, label=self.label)
 
     def _define(self) -> None:
         qc = qiskit.QuantumCircuit(2, name=self.name)
-        first_sign = +1 if self.polarity == "+-" else -1
-        qc.rzx(first_sign * np.pi / 4, 0, 1)
+        if self.rads == 0:
+            self.definition = qc
+            return
+
+        qc.rzx(self.rads / 2, 0, 1)
         qc.x(0)
         if self.sandwich_rx_rads:
             qc.rx(self.sandwich_rx_rads, 1)
-        qc.rzx(-first_sign * np.pi / 4, 0, 1)
+        qc.rzx(-self.rads / 2, 0, 1)
         self.definition = qc
 
     def __array__(self, dtype: Optional[type] = None) -> npt.NDArray[np.bool_]:
-        cval = 1 / np.sqrt(2)
-        if self.polarity == "+-":
-            sval = 1j * cval
-        else:
-            sval = -1j * cval
-
-        mat = np.array(
-            [
-                [0, cval, 0, sval],
-                [cval, 0, -sval, 0],
-                [0, sval, 0, cval],
-                [-sval, 0, cval, 0],
-            ],
-            dtype=dtype,
-        )
-
-        # sandwiched rx gate commutes and can just be multiplied with non-sandwiched part:
-        return mat @ np.kron(
-            np.asarray(qiskit.circuit.library.RXGate(self.sandwich_rx_rads), dtype=dtype),
-            np.eye(2, dtype=dtype),
-        )
+        return qiskit.quantum_info.Operator(self.definition).data
 
     def __repr__(self) -> str:
-        args = f"'{self.polarity}'"
-        if self.sandwich_rx_rads:
-            args += f", sandwich_rx_rads={self.sandwich_rx_rads}"
+        args = []
+        if self.rads not in [0, np.pi / 2]:
+            args.append(f"rads={self._rads_str}")
```

Comment:
```suggestion
            args.append(f"rads={self.rads!r}")
```
(repr values shouldn't be rounded/include the unicode pi)